skyfield/positionlib.py

```diff
@@ -152,18 +151,61 @@ def separation_from(self, another_icrf):
         c = dots(u1, u2)
         return Angle(radians=arccos(clip(c, -1.0, 1.0)))
 
-    def ecliptic_position(self):
-        """Compute J2000 ecliptic coordinates (x, y, z)"""
-        vector = _ECLIPJ2000.dot(self.position.au)
-        return Distance(vector)
+    def ecliptic_xyz(self, epoch=None):
+        """Compute J2000 ecliptic coordinates (x, y, z)
 
-    def ecliptic_latlon(self):
-        """Compute J2000 ecliptic coordinates (lat, lon, distance)"""
-        vector = _ECLIPJ2000.dot(self.position.au)
-        d, lat, lon = to_polar(vector)
-        return (Angle(radians=lat, signed=True),
-                Angle(radians=lon),
-                Distance(au=d))
+        If you instead want the coordinates referenced to the dynamical
+        system defined by the Earth's true equator and equinox, provide
+        an epoch time.
+        """
+        lat, lon, d = self.ecliptic_latlon(epoch)
+        vector = from_polar(d.au, lat.radians, lon.radians)
+        if len(vector.shape) is 1:
+            return Distance(vector)
+        else:
+            return Distance(vector.T)
+
+    def ecliptic_latlon(self, epoch=None):
+        """Compute J2000 ecliptic coordinates (lat, lon, distance)
+
+        If you instead want the coordinates referenced to the dynamical
+        system defined by the Earth's true equator and equinox, provide
+        an epoch time.
+        """
+        position_au = self.position.au
+        if epoch is not None:
+            if isinstance(epoch, Time):
+                pass
+            elif isinstance(epoch, float):
+                epoch = Time(None, tt=epoch)
+            elif epoch == 'date':
+                epoch = self.t
+            else:
+                raise ValueError('the epoch= must be a Time object,'
+                                 ' a floating point Terrestrial Time (TT),'
+                                 ' or the string "date" for epoch-of-date')
+            position_au = einsum('ij...,j...->i...', epoch.M, position_au)
+            if not epoch.shape:
+                vector = epoch.E.dot(position_au)
+                d, lat, lon = to_polar(vector)
+                return (Angle(radians=lat, signed=True),
+                        Angle(radians=lon),
+                        Distance(au=d))
+            else:
+                result_array = empty((epoch.E.T.shape[0], epoch.E.T.shape[1]))
+                for a in range(0, epoch.E.T.shape[0]):
+                    vector = epoch.E.T[a, 0:].dot(position_au.T[a, 0:])
+                    result_array[a, 0:] = vector
+                d, lat, lon = to_polar(result_array.T)
+                return (Angle(radians=lat, signed=True),
+                        Angle(radians=lon),
+                        Distance(au=d))
+        else:
+            vector = _ECLIPJ2000.dot(position_au)
```

Comment:
Nice! I like how you kept the routine compatible with its old behavior if `epoch` stays None.

I wonder if we could eliminate `_ECLIPJ2000` from the code, if the `earth_tilt()` routine for J2000 returns exactly the same angle as `_ECLIPJ2000` is built with?

Reply:
I will try to figure it out, yes it would be nice to clean it up.

Reply:
Ok so I experimented with it. If I call earth_tilit for t.tdb(2000,1,1,12,0,0) which is exactly J2000,
and then I form the transformation matrix, I get this
[[1.0,  0.0,                                   0.0],
 [0.    0.9174932688587917,  0.3977513062188596],
 [0.   -0.3977513062188596,  0.9174932688587917]]
compared to what we already have:
[[1.0,  0.0,                                      0.0],
 [0.0,  0.91748206206918181, 0.39777715593191371],
 [0.0, -0.39777715593191371, 0.91748206206918181]]

I am using mean obliquity, because for some crazy reason it seems to be closer to the matrix that we already use. I can eliminate _ECLIPJ2000 if you are OK with using these values. I am not sure what do we gain by introducing unnecessary calculations though compared to fixed value that is prepared.